samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlExecutionContext.java

```diff
@@ -52,16 +54,19 @@ private SamzaSqlExecutionContext(SamzaSqlExecutionContext other) {
 
   public SamzaSqlExecutionContext(SamzaSqlApplicationConfig config) {
     this.sqlConfig = config;
-    udfMetadata =
-        this.sqlConfig.getUdfMetadata().stream().collect(Collectors.toMap(UdfMetadata::getName, Function.identity()));
+    udfMetadata = new HashMap<>();
+    for(UdfMetadata udf : this.sqlConfig.getUdfMetadata()) {
+      udfMetadata.putIfAbsent(udf.getName(), new ArrayList<>());
+      udfMetadata.get(udf.getName()).add(udf);
+    }
   }
 
   public ScalarUdf getOrCreateUdf(String clazz, String udfName) {
     return udfInstances.computeIfAbsent(udfName, s -> createInstance(clazz, udfName));
   }
 
   public ScalarUdf createInstance(String clazz, String udfName) {
-    Config udfConfig = udfMetadata.get(udfName).getUdfConfig();
+    Config udfConfig = udfMetadata.get(udfName).get(0).getUdfConfig();
```

Comment:
Here, is it possible to move "[private final Config udfConfig](https://github.com/apache/samza/blob/c6ca30cc4869548a7766848c90833b4c133d6282/samza-sql/src/main/java/org/apache/samza/sql/interfaces/UdfMetadata.java#L38)" from UdfMetadata (SamzaSQLUDFMethod level) to UDF level?

Reply:
Good catch!! UdfMetadata is associated at the udflevel, We could consider creating a class level and then method level udf metadata. Creating such a structure just for holding the config seemed unncessary. I will add the comment here clarifying this behavior.